### ISSUE TITLE
3.19.0 - Post-Quantum VPN question

### ISSUE THREAD

**Issue by ph202107** [NordSecurity/nordvpn-linux#637]
Hi!  I'm Using NordVPN 3.19.0 on Linux Mint 21.3.

Does the post-quantum setting need to be enabled during the initial handshake and key exchange?

For example if I'm already connected to a VPN server (using NordLynx), and while connected I enable the post-quantum setting, do I then need to reconnect to the VPN server to perform another handshake?

I could not find any information on the support site, blog site, man nordvpn, or --help.  In the nordvpnd.service logs there seems to be only one entry when the setting is changed:  "PostquantumVpn set to: enabled"

Thanks for any clarification.

**Comment by ph202107** [comment]
I can rephrase.

Does the post-quantum setting need to be enabled BEFORE connecting to VPN?

If the post-quantum setting is enabled AFTER the VPN connection is already established, is post-quantum encryption being utilized immediately, or only after the VPN reconnects?

I contacted Nord Support via the live chat but they have no information about this new feature.

**Comment by keliramu** [comment]
Hello, post-quantum setting should be turned on before connecting to VPN, and if you are already connected to VPN and change post-quantum setting - it will take effect only on next connect or on reconnect.

**Comment by ph202107** [comment]
Hi keliramu, thanks for the clarification.

Perhaps a message similar to this one could be helpful to let the user know that post-quantum encryption is not actually in effect until the VPN reconnects:

```
$ nordvpn set technology openvpn
Technology is set to 'OPENVPN' successfully.
You are connected to NordVPN. Please reconnect to enable the setting.
$
```

**Comment by keliramu** [comment]
Hi, thank you for your input, you are right - and we are planning to improve this with next release.

**Comment by ph202107** [comment]
Resolved with release 3.19.1.  Thanks!
